File: src/org/espenhahn/allocate/likepaxos/registry/debug/mvc/animations/GrowLineAnimation.java
package org.espenhahn.allocate.likepaxos.registry.debug.mvc.animations;

import java.util.function.Consumer;

import org.espenhahn.allocate.likepaxos.registry.debug.mvc.objects.Line;

public class GrowLineAnimation extends TranslateAnimation<Line> {

	public GrowLineAnimation(Line l, int end_x, int end_y, Consumer<Line> callback) {
		super(l, end_x, end_y, callback);
	}
	
	@Override
	protected void apply(int dx, int dy) {
		getDrawable().translateDelta(dx, dy);
	}

}

File: src/org/espenhahn/allocate/likepaxos/registry/debug/mvc/animations/GrowLineAnimation.kt
package org.espenhahn.allocate.likepaxos.registry.debug.mvc.animations

import org.espenhahn.allocate.likepaxos.registry.debug.mvc.objects.Line
import java.util.function.Consumer

class GrowLineAnimation(l: Line, end_x: Int, end_y: Int, callback: Consumer<Line>) : TranslateAnimation<Line>(l, end_x, end_y, callback) {

    override fun apply(x: Int, y: Int) {
        drawable.translateDelta(x, y)
    }

}
